app/server/lib/FlexServer.ts

```diff
@@ -901,11 +915,18 @@ export class FlexServer implements GristServer {
 
     // TODO: We could include a third mock provider of login/logout URLs for better tests. Or we
     // could create a mock SAML identity provider for testing this using the SAML flow.
-    const loginSystem = await (process.env.GRIST_TEST_LOGIN ? getTestLoginSystem() : getLoginSystem());
+    const loginSystem = await (process.env.GRIST_TEST_LOGIN ? getTestLoginSystem() :
+      (this._getLoginSystem?.() || getLoginSystem()));
     this._loginMiddleware = await loginSystem.getMiddleware(this);
     this._getLoginRedirectUrl = tbind(this._loginMiddleware.getLoginRedirectUrl, this._loginMiddleware);
     this._getSignUpRedirectUrl = tbind(this._loginMiddleware.getSignUpRedirectUrl, this._loginMiddleware);
     this._getLogoutRedirectUrl = tbind(this._loginMiddleware.getLogoutRedirectUrl, this._loginMiddleware);
+    if (this._loginMiddleware.getWildcardMiddleware) {
+      const wildcardMiddleware = this._loginMiddleware.getWildcardMiddleware();
+      if (wildcardMiddleware.length > 0) {
+        this.app.use(wildcardMiddleware);
+      }
+    }
```

Comment:
```suggestion
    if (this._loginMiddleware.getWildcardMiddleware?.().length > 0) {
      this.app.use(wildcardMiddleware);
    }
```
Condition is a bit longer but the body is shorter, if you prefer the tradeoff.

Reply:
Used a slightly tweaked version, thanks.